lib/cache/constants/cache-constants.ts

```diff
@@ -2,8 +2,8 @@
 
 export class CacheConstants {
   public static LABSHARE_CACHE = 'LABSHARE_CACHE';
-  public static REDIS_OPTIONS = 'redisOptions';
-  public static CACHE_STRATEGY = 'strategy';
+  public static REDIS_SETTINGS = 'redis';
+  public static CACHE_TYPE = 'type';
```

Comment:
What's the purpose of having this and other constants like this? Their values are essentially names of the cache config properties which never change.  
IMO there should be `types.ts` file which declares a structure of cache config object as well as an enum for cache types. Something like this: 

```TypeScript
export class CacheConfig {
    type: string;
    redis?: RedisOptions;
    memory?: MemoryOptions; 
}
export enum CacheType {
   REDIS = 'redis',
   MEMORY = 'memory'
} 
```